### PR TITLE
Add utils for RouteManager and modify its internal data.

### PR DESCRIPTION
**Pull request by siarheishut**
Maintain maps 'stop->stop_id' and 'stop_id->stop' and add functions to calculate vertex statuses instead of maintaining 'stop->vertex_ids'. 
Create rm::RouteManager namespace for RouteManager class and its utils.

### CODE REVIEW

File: src/route_manager.cpp [hunk 1]
```diff
@@ -8,32 +8,53 @@
...
-      vertices_(stop_info.size() * 2) {
+      stop_names_(stop_info.size()) {
   ReadStops(stop_info);
   ReadBuses(bus_info, stop_info);
   router_ = std::make_unique<Router>(graph_);
 }
 
 void RouteManager::ReadStops(const StopDict &stop_dict) {
-  int vertex_id = 0;
+  size_t stop_idx = 0;
```
Comment (foxfrikses) [inline src/route_manager.cpp:51]: stop_id instead of stop_idx

File: src/route_manager.cpp [hunk 2]
```diff
@@ -8,32 +8,53 @@
...
   for (auto &[stop, _] : stop_dict) {
-    auto &[arrive, depart] = stop_ids_[stop];
-    arrive = vertex_id++;
-    depart = vertex_id++;
-    vertices_[arrive] = Vertex{stop};
-    vertices_[depart] = Vertex{stop};
+    stop_ids_[stop] = stop_idx++;
```
Comment (foxfrikses) [inline src/route_manager.cpp:53]: It's easier to read if you do increments on separate lines

File: src/route_manager.cpp [hunk 3]
```diff
@@ -8,32 +8,53 @@
...
+    stop_names_[stop_ids_[stop]] = stop;
```
Comment (foxfrikses) [inline src/route_manager.cpp:54]: If you move increments to after this line, you will be able to replace stop_ids_[stop] with stop_idx

File: src/route_manager.cpp [hunk 4]
```diff
@@ -8,32 +8,53 @@
...
 
-namespace rm {
+namespace {
+using namespace rm::route_manager;
+bool IsArrive(size_t vertex) {
+  return (vertex % 2 == 0);
+}
+
+bool IsDepart(size_t vertex) {
+  return (vertex % 2 == 0);
```
Comment (foxfrikses) [inline src/route_manager.cpp:22]: Why is arrive and is depart are identical?

File: src/route_manager.cpp [hunk 5]
```diff
@@ -8,32 +8,53 @@
...
-    arrive = vertex_id++;
-    depart = vertex_id++;
-    vertices_[arrive] = Vertex{stop};
-    vertices_[depart] = Vertex{stop};
+    stop_ids_[stop] = stop_idx++;
+    stop_names_[stop_ids_[stop]] = stop;
     graph_.AddEdge({
-                       .from = arrive,
-                       .to = depart,
+                       .from = ToArrive(stop_ids_[stop]),
```
Comment (foxfrikses) [inline src/route_manager.cpp:56]: If you move increment after this line, you'll be able to replace stop_ids_[stop] with stop_idx

File: src/route_manager.cpp (hunk 4 above)
Comment (siarheishut) [reply src/route_manager.cpp:22]: Fixed.

File: src/bus_manager.h [hunk 6]
```diff
@@ -28,7 +28,7 @@ class BusManager {
 
  private:
   std::shared_ptr<TransportCatalog> transport_catalog_;
-  std::unique_ptr<RouteManager> route_manager_;
+  std::unique_ptr<route_manager::RouteManager> route_manager_;
```
Comment (foxfrikses) [inline src/bus_manager.h:31]: I think the RouteManager can be left in rm namespace: `route_manager::route_manager` doesn't seem to add more clarity.

File: src/route_manager.cpp [hunk 7]
```diff
@@ -8,34 +8,56 @@
 #include "common.h"
 #include "distance_computer.h"
 #include "request_types.h"
+#include "route_manager_utils.h"
 
 using namespace rm::utils;
 
-namespace rm {
+namespace {
+using namespace rm::route_manager;
```
Comment (foxfrikses) [inline src/route_manager.cpp:16]: Why is `using namespace rm::route_manager;` inside an unknown namespace?

**Review by foxfrikses — APPROVED** [review]
LGTM

File: src/route_manager.cpp (hunk 7 above)
Comment (siarheishut) [reply src/route_manager.cpp:16]: Moved it from unnamed namespace.

**Pull request merged**